src/components/UserSelector.tsx

```diff
@@ -1,33 +1,69 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type Props = {
+  users: User[],
+  selectedUserId: number,
+  onSelectUser: (userId: number) => void,
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  selectedUserId,
+  onSelectUser,
+}) => {
+  const [usersAreVisible, setUsersAreVisible] = useState(false);
 
-export const UserSelector: React.FC = () => {
   return (
     <div
       data-cy="UserSelector"
-      className="dropdown is-active"
+      className={classNames(
+        'dropdown',
+        { 'is-active': usersAreVisible },
+      )}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setUsersAreVisible(!usersAreVisible)}
+          onBlur={() => setUsersAreVisible(false)}
         >
-          <span>Choose a user</span>
+          <span>
+            {users.find(user => user.id === selectedUserId)?.name
+              || 'Choose a user'}
+          </span>
 
           <span className="icon is-small">
             <i className="fas fa-angle-down" aria-hidden="true" />
           </span>
         </button>
       </div>
 
-      <div className="dropdown-menu" id="dropdown-menu" role="menu">
+      <div
+        className="dropdown-menu"
+        id="dropdown-menu"
+        role="menu"
+      >
         <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">Leanne Graham</a>
-          <a href="#user-2" className="dropdown-item is-active">Ervin Howell</a>
-          <a href="#user-3" className="dropdown-item">Clementine Bauch</a>
-          <a href="#user-4" className="dropdown-item">Patricia Lebsack</a>
-          <a href="#user-5" className="dropdown-item">Chelsey Dietrich</a>
+          {users.map(user => (
+            <a
+              href="#user-1"
```

Comment:
Change the href attribute value to `#user-${user.id}` to make it dynamic and unique for each user.